Generic TSRs

jeking04 opened the pull request:
This implements cylinder and box TSRs for use by the tsr library.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+    ''' 
+    Helper function for getting the manipulator index
+    to be used by the TSR
+    '''
```
Add docstrings for arguments. Also use `"""` instead of `'''` for multi-line comments.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+                   vertical_tolerance = 0.02,
+                   yaw_range = None,
+                   manip = None, **kwargs):
+    """
```
Add a one-line description of the function to the docstring. It currently lists assumptions, but does not explain what the function *does*.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+    @param yaw_range Allowable range of yaw around object (default: [-pi, pi])
+    @param manip The manipulator to perform the grasp, if None
+       the active manipulator on the robot is used
+    """
```
Add bounds checking for `obj_radius`, `obj_height`, `vertical_tolerance`, and `yaw_range`.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+    NOTE: This function makes the following assumptions:
+      1. The end-effector is oriented such that the z-axis is out of the palm
+          and the x-axis should be perpendicular to the object
+      2. The object coordinate frame is at the bottom, center of the object
```
Add a one-line description of this function.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+      a good grasp
+    @param manip The manipulator to perform the grasp, if None the active 
+      manipulator on the robot is used
+    """
```
Add the same error checking I described above.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+    ee_in_obj = numpy.dot(numpy.linalg.inv(obj.GetTransform()), 
+                          manip.GetEndEffectorTransform())
+    Bw = numpy.zeros((6,2)) 
+   
```
Verify that the object is indeed grabbed by the correct manipulator.

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+                      pitch_epsilon=0.2, 
+                      yaw_epsilon=0.2,
+                      manip=None, **kwargs):
+    '''
```
`"""` not `'''`

mkoval commented on src/prpy/tsr/generic.py:
```diff
@@ -0,0 +1,275 @@
+    @param yaw_epsilon The amount to let the object yaw during transport (object frame)
+    @param manip the manipulator grasping the object, if None the active manipulator 
+       of the robot is used
+    '''
```
Check bounds on `roll_epsilon`, `pitch_epsilon`, and `yaw_epsilon`.

jeking04 commented:
Alrighty @mkoval, I believe I have addressed all comments. I am just waiting for the checks to pass, but you can go ahead and have another look whenever convenient.
